zowex/src/main.rs

```diff
@@ -408,15 +709,26 @@ fn talk(message: &[u8], stream: &mut DaemonClient) -> io::Result<()> {
                     u_payload.pop(); // remove the 0xC
                     payload = str::from_utf8(&u_payload).unwrap().to_string();
 
-                    let p: DaemonRequest = match serde_json::from_str(&payload) {
+                    let p: DaemonRequest;
+                    match serde_json::from_str(&payload) {
                         Err(_e) if _progress => {
                             if atty::is(Stream::Stderr) {
                                 eprint!("{}", payload);
                                 io::stderr().flush().unwrap();
                             }
                             continue;
                         },
-                        result => result.unwrap(),
+                        result => {
+                            match result {
+                                Ok(ok_val) => {
+                                    p = ok_val;
+                                },
+                                Err(err_val) => {
+                                    eprint!("You may be running mismatched versions of Zowe executable and Zowe daemon.\n");
```

Comment:
Clippy (Rust linter) doesn't like this line: ``warning: using `eprint!()` with a format string that ends in a single newline``

Reply:
I forgot to run clippy once more after my last change. All I knew was that the next line of output after the eprint statement was concatenated onto the end of the eprint output line. Adding the newline to the eprint statement seemed like the obvious fix. I did not even know that 'eprintln' was a thing. This was a good thing to change.

Reply:
I agree that the file size is big, but getting Rust to compile is so darn hard that I am leery of undertaking that now. We can keep the idea in mind when we undertake another significant change to Rust code. Maybe it could be done in the upcoming fix for detecting the right daemon?